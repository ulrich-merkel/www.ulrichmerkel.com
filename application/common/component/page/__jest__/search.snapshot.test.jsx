/* eslint-disable import/no-extraneous-dependencies, func-names */
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import mockedStore from './../../__mocks__/store';
import PageSearch from './../search';

describe('common/component/page/search', function () {
    const props = {
        content: {
            head: {},
            section1: {},
            formSearch: {}
        }
    };

    it('should render correctly', function () {
        const tree = renderer.create(
            <Provider store={mockedStore}>
                <PageSearch {...props}>
                    Page search children
                </PageSearch>
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('should render correctly as dialog', function () {
        const tree = renderer.create(
            <Provider store={mockedStore}>
                <PageSearch {...props} isDialog>
                    Page search children as dialog
                </PageSearch>
            </Provider>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
