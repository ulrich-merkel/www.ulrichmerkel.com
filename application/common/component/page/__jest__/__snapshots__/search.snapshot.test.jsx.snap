// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/component/page/search should render correctly 1`] = `
<main
  className="l-main"
  itemProp="mainContentOfPage"
  role="main"
>
  <div
    className="l-main__transition"
  >
    <section
      className="l-grid m-section"
    >
      <div
        className="l-grid__spaced--small"
      >
        <div
          className="l-grid__row"
        >
          <div
            className="l-grid__col--12"
          >
            <article
              className="is-main m-article"
              itemScope={true}
              itemType="https://schema.org/Article"
            >
              <div
                className="m-article__text"
                itemProp="text"
              >
                <form
                  acceptCharset="utf-8"
                  action="/search/"
                  className="m-form m-form--search"
                  id="m-form--search"
                  itemProp="potentialAction"
                  itemScope={true}
                  itemType="http://schema.org/SearchAction"
                  method="post"
                  noValidate={true}
                  onReset={[Function]}
                  onSubmit={[Function]}
                  role="search"
                >
                  <fieldset
                    className="m-form__fieldset"
                  >
                    <legend
                      className="m-form__legend is-visually-hidden"
                    />
                    <div
                      className="l-grid__row"
                    >
                      <div
                        className="l-grid__col--12"
                      >
                        <div
                          className="m-form__group"
                        >
                          <input
                            aria-required={true}
                            className="m-form__control--text"
                            id="name"
                            itemProp="query-input"
                            name="name"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder=""
                            required={true}
                            type="search"
                            value=""
                          />
                          <label
                            className="m-form__label is-visually-hidden"
                            htmlFor="name"
                          >
                            
                            <i
                              aria-hidden="true"
                              className="c-font-icon--sad m-form__label__icon--error"
                            />
                            <i
                              aria-hidden="true"
                              className="c-font-icon--smile m-form__label__icon--success"
                            />
                          </label>
                        </div>
                      </div>
                    </div>
                    <input
                      name="_csrf"
                      type="hidden"
                      value=""
                    />
                  </fieldset>
                </form>
                <h3
                  className="is-centered c-type--h3"
                >
                  Found 0 search results
                </h3>
              </div>
            </article>
          </div>
        </div>
      </div>
    </section>
  </div>
</main>
`;

exports[`common/component/page/search should render correctly as dialog 1`] = `
<div
  className="l-main__transition"
>
  <section
    className="l-grid m-section"
  >
    <div
      className="l-grid__spaced--small"
    >
      <div
        className="l-grid__row"
      >
        <div
          className="l-grid__col--12"
        >
          <article
            className="is-main m-article"
            itemScope={true}
            itemType="https://schema.org/Article"
          >
            <div
              className="m-article__text"
              itemProp="text"
            >
              <form
                acceptCharset="utf-8"
                action="/search/"
                className="m-form m-form--search"
                id="m-form--search"
                itemProp="potentialAction"
                itemScope={true}
                itemType="http://schema.org/SearchAction"
                method="post"
                noValidate={true}
                onReset={[Function]}
                onSubmit={[Function]}
                role="search"
              >
                <fieldset
                  className="m-form__fieldset"
                >
                  <legend
                    className="m-form__legend is-visually-hidden"
                  />
                  <div
                    className="l-grid__row"
                  >
                    <div
                      className="l-grid__col--12"
                    >
                      <div
                        className="m-form__group"
                      >
                        <input
                          aria-required={true}
                          className="m-form__control--text"
                          id="name"
                          itemProp="query-input"
                          name="name"
                          onBlur={[Function]}
                          onChange={[Function]}
                          placeholder=""
                          required={true}
                          type="search"
                          value=""
                        />
                        <label
                          className="m-form__label is-visually-hidden"
                          htmlFor="name"
                        >
                          
                          <i
                            aria-hidden="true"
                            className="c-font-icon--sad m-form__label__icon--error"
                          />
                          <i
                            aria-hidden="true"
                            className="c-font-icon--smile m-form__label__icon--success"
                          />
                        </label>
                      </div>
                    </div>
                  </div>
                  <input
                    name="_csrf"
                    type="hidden"
                    value=""
                  />
                </fieldset>
              </form>
              <h3
                className="is-centered c-type--h3"
              >
                Found 0 search results
              </h3>
            </div>
          </article>
        </div>
      </div>
    </div>
  </section>
</div>
`;
